src/show.jl

```diff
@@ -33,7 +33,7 @@ function _use_multline_show(d::Distribution, pnames)
     tlen = 0
     for (i, p) in enumerate(pnames)
         pv = getfield(d, p)
-        if !(isa(pv, Number) || isa(pv, NTuple) || isa(pv, AbstractVector))
+        if isa(pv, NTuple) || isa(pv, Array)
```

Comment:
@bdeonovic I am a bit confused by this change - it seems it should rather be
```suggestion
         if pv isa AbstractMatrix
```
to correspond exactly to the criterion mentioned in the comment above. But in general, it seems that this hardcoded criterion is a bit unflexible - I assume that there are many other more complex parameter types which should rather be printed in multiple lines. I guess this motivated the previous implementation which instead defined a whitelist of types that can be printed in one line.

BTW I came across this PR because it seems this change leads to errors in https://github.com/JuliaGaussianProcesses/AbstractGPs.jl/pull/80.

Reply:
probably meant to tag @briochemc

Reply:
Oh sorry, indeed this was my intention :grimacing:

Reply:
I agree? 🤷 FWIW, I think I merely meant to remove the !isa(pv, Number)...

I guess your suggestion is cleaner and simpler, too!

Reply:
IMO it would be better to change the comment and revert the implementation. The change was quite breaking for more complex distributions (such as `AbstractGPs.FiniteGP`) since now the default is that `length` has to be implemented for all fields that are not a tuple or an array. This would not be resolved by checking `pv isa AbstractMatrix`.

Reply:
I think I get it. Now that I think of it, I think this only worked here with `LocationScale` because univariate distributions come with a length of 1. I guess a better solution would be to specialize `show` for `LocationScale` instead?

Reply:
The problem is that any distribution with a field that is neither a `NTuple` nor an `Array` and does not implement `length(::typeof(field))` (which is called in the other branch) is broken currently. It can even happen for univariate distributions. So I think it would be best to just revert this change and release a new version as soon as possible.

If the update for `LocationScale` can't be integrated in a non-breaking way, it might be best to specialize `show`, as you suggest.

Reply:
Oh yes, I meant that in a LocationScale, one of the fields is a univariate distribution, of which the length is 1. So that worked by chance as I don’t think that was the intended use.